extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -55,183 +62,356 @@
+            if (!seq.isEmpty() && seq.getItemType() != Type.STRING) {
+                throw new XPathException(this, ErrorCodes.XPTY0004,
+                        "Invalid value for option \"excludes\", expected xs:string* got " +
+                                Type.getTypeName(seq.getItemType()));
+            }
+            options.put(EXCLUDES_OPT, seq);
+
+            checkOption(optionsMap, PRUNE_OPT, Type.BOOLEAN, options);
+            checkOption(optionsMap, AFTER_OPT, Type.DATE_TIME, options);
+        } else if (parameter.itemAt(0).getType() == Type.DATE_TIME) {
+            // QUESTION(JL) LOG deprecation warning?
+            options.put(AFTER_OPT, parameter);
+        } else {
+            throw new XPathException(this, ErrorCodes.XPTY0004,
+                    "Invalid 3rd parameter, allowed are xs:dateTime or map(*) got " + Type.getTypeName(item.getType()));
+        }
+        return options;
+    }
+
+    private void checkOption(
+            final AbstractMapType optionsMap,
+            final String name,
+            final int expectedType,
+            final HashMap<String, Sequence> options
+    ) throws XPathException {
+        final Sequence p = optionsMap.get(new StringValue(name));
+
+        if (p.isEmpty()) {
+            return; // nothing to do, continue
+        }
+
+        if (p.hasMany() || p.getItemType() != expectedType) {
+            throw new XPathException(this, ErrorCodes.XPTY0004,
+                    "Invalid value for option \"" + name + "\", expected " +
+                            Type.getTypeName(expectedType) + " got " +
+                            Type.getTypeName(p.itemAt(0).getType()));
+        }
+
+        options.put(name, p);
+    }
+
+    private Sequence startSync(
+            final String target,
+            final String collectionPath,
+            final HashMap<String, Sequence> options
+    ) throws XPathException {
+        final Date startDate = options.get(AFTER_OPT).hasOne() ? ((DateTimeValue) options.get(AFTER_OPT)).getDate() : null;
+
+        boolean prune = ((BooleanValue) options.get(PRUNE_OPT)).getValue();
+
+        final List<String> excludes = new ArrayList<>(Collections.emptyList());
+        for (final SequenceIterator si = options.get(EXCLUDES_OPT).iterate(); si.hasNext(); ) {
+            excludes.add(si.nextItem().getStringValue());
+        }
+
+        final Path p = FileModuleHelper.getFile(target);
+        context.pushDocumentContext();
+        final MemTreeBuilder output = context.getDocumentBuilder();
+        final Path targetDir;
+        try {
+            if (p.isAbsolute()) {
+                targetDir = p;
+            } else {
+                final Optional<Path> home = context.getBroker().getConfiguration().getExistHome();
+                targetDir = FileUtils.resolve(home, target);
+            }
+
+            output.startDocument();
+            output.startElement(Sync.FileSyncElement, null);
```

Comment:
The static-final object members don't need to be be prefixed `Sync.` as they are already in scope.